azure/services/disks/disks.go

```diff
@@ -38,16 +39,20 @@ type DiskScope interface {
 // Service provides operations on Azure resources.
 type Service struct {
 	Scope DiskScope
-	async.Reconciler
+	asyncpoller.Reconciler
 }
 
-// New creates a new disks service.
-func New(scope DiskScope) *Service {
-	client := newClient(scope)
-	return &Service{
-		Scope:      scope,
-		Reconciler: async.New(scope, nil, client),
+// New creates a disks service.
+func New(scope DiskScope) (*Service, error) {
+	client, err := newClient(scope)
+	if err != nil {
+		return nil, err
 	}
+	return &Service{
+		Scope: scope,
+		Reconciler: asyncpoller.New[armcompute.DisksClientCreateOrUpdateResponse,
+			armcompute.DisksClientDeleteResponse](scope, nil, client),
```

Comment:
There's no `CreateOrUpdate` functionality in this client, so although it has to be typed specifically to satisfy the generic interface, the Creator is `nil`.